controlplane/kubeadm/controllers/upgrade.go

```diff
@@ -82,6 +82,12 @@ func (r *KubeadmControlPlaneReconciler) upgradeControlPlane(
 		}
 	}
 
+	if kcp.Spec.KubeadmConfigSpec.ClusterConfiguration != nil {
```

Comment:
Should we do a zero value check for `kcp.Spec.KubeadmConfigSpec.ClusterConfiguration.APIServer` to avoid accidental overrides with empty configuration?

Reply:
cc @fabriziopandini

Reply:
I don't think we should implement a zero check given that zero/empty configuration is equal to "use the kubeadm defaults"

Expanding a little bit,  think we should make sure to support moving from zero/kubeadm defaults to kubeadm defaults U extra settings and also the vice-versa. eg.
- I'm using kubeadm defaults (zero/empty)
- I temporarily enable logging with an extra arg
- I disable logging by removing the extra args